dbms/src/Databases/DatabaseMySQL.h

```diff
@@ -0,0 +1,125 @@
+#pragma once
+
+#include <Common/config.h>
+
+#if USE_MYSQL
+
+#include <mysqlxx/Pool.h>
+#include <Databases/DatabasesCommon.h>
+
+namespace DB
+{
+
+/** Real-time access to table list and table structure from remote MySQL
+ *  It doesn't make any manipulations with filesystem.
+ *  All tables are created by calling code after real-time pull-out structure from remote MySQL
+ */
+class DatabaseMySQL : public IDatabase
```

Comment:
BTW, how it was working without cache?

Reply:
The current way of working is not very efficient. Currently, a simple query(no join, no subquery) will call the `tryGetStorage` method five or six times.

The MySQL database engine attempts to query MySQL system tables every time `tryGetStorage` is called. This will exist the following scenarios:
- exist in cache & no exist in mysql : `Move tables to outdated list for asynchronous drop.`
- no exist in cache & exist in mysql : `Create tables and into it to cache.`
- exist in cache & exist in mysql, but table structure are updated: `Execute both cases simultaneously`

Reply:
I'm not sure what would happen if you returned different storage to the same query. Maybe we should write a test ?

Reply:
Ok, the cache is reasonable.